designs/inter-ident-verify.md

```diff
@@ -0,0 +1,78 @@
+___
+## Proposed solution
+Many companies have various branches that operate different parts of the company. For example, take the following parties:
+* U, the individual user
+* A, the big name corporation the user knows of
+* S, a branch of A that handles tech support, and is trusted by A
+
+If A wanted to contact U, the CEO wouldn't call them directly. Instead, they would reach out through S, which lives in a tightly vetted whitelist of callers approved by A. S calls U, and U must verify that S is a trusted branch of A, with only the knowledge that S supposedly is trusted by A. U doesn't care about the exact identity of the caller, only that they are trusted by A, since no scam company could make A's tightly vetted whitelist of trusted callers. Since they're not contacting A directly, there are four key connections that must be verified:
+1. U needs to know the S is trusted by A.
+2. A needs to verify the S is trusted.
+3. U must verify their request for verification was directly verified by A and nobody else.
+4. U needs to know the response received is not a replay attack, and that the response received is unique to their request for verification.
+
+This can be accomplished by sending a nonce encrypted message to A, and then sending it back to U with a different encryption pattern. If the message received is the same as the message sent, then all identities are verified. Since the message sent is always unique, replay attacks are meaningless.
+
+### Kanidm Server setup
+In this situation, there are two Kanidm instances which can be remote from each other:
+* X, local to A
+* Y, local to S and U
+
+All users (individuals, companies, company branches, etc.) of this verification system will have an existing RSA key linked to their Kanidm credentials on their local servers. This key will not change (unless for very good reason), and the private key is only known by the exact account without exception. Since these keys are linked to credentials, we can safely find the authentic public key of any user we want, without real time communication between servers around the world. Additionally, all trusted callers for A will also have a copy of their information (name, public key) linked to A's credentials on server X. This is acceptable since there are limited trusted callers for A. Furthermore, server Y will store a copy of A's information (name, public key) locally. This is acceptable, since there are limited large corporations. It's also important to note that A doesn't need to store any information about any particular U on server X.
```

Comment:
When you say "on their local server" do you mean, "stored as an attribute in their own account entry on the kanidm server"? Or is the key stored outside of the kanidm instance somehow?

Also please use ECDSA or ED25519 instead of RSA :)